std/format.d

```diff
@@ -3040,19 +3040,49 @@ if (isInputRange!T)
             static if (is(StringTypeOf!T))
             {
                 auto s = val[0 .. f.precision < $ ? f.precision : $];
+
+                // Avoid expensive call to byGrapheme
+                bool onlyAscii;
+                static if (is(CharTypeOf!E == char))
+                {
+                    onlyAscii = true;
+                    for (size_t i; i < s.length; i++)
+                        if (s[i] > 0x7F)
+                    {
+                        onlyAscii = false;
+                        break;
+                    }
+                }
+
+                size_t width;
+                if (!onlyAscii)
+                {
+                    // FIXME: byGrapheme.walkLength cannot be used in CTFE
+                    // see https://issues.dlang.org/show_bug.cgi?id=18209
+                    import std.uni : byGrapheme, graphemeStride;
+                    if (__ctfe)
+                    {
+                        size_t stride;
+                        for (size_t i = 0; i < s.length; i += graphemeStride(s, i))
+                            width++;
+                    }
+                    else width = val.byGrapheme.walkLength;
```

Comment:
We don't really need to use `byGrapheme` at all, since we don't care about the individual `Grapheme` structs. The loop in the `__ctfe` branch works equally well in CTFE and at runtime. Just use the `graphemeStride` loop always, no need to distinguish between CTFE and runtime.